src/lib/app/versions.tsx

```diff
@@ -70,7 +69,64 @@ export async function getLatestAvailableAppVersionForApp(
   );
 }
 
-// App Version UUID from App Version irellevant if internal or external
+export async function getAllUpgradeCandidatesFromAppInstallationId(
+  apiClient: MittwaldAPIV2Client,
+  appInstallationId: string,
+): Promise<AppVersion[]> {
+  const currentAppInstallation = await getAppInstallationFromUuid(
+      apiClient,
+      appInstallationId,
+    ),
+    updateCandidates = await apiClient.app.listUpdateCandidatesForAppversion({
+      appId: currentAppInstallation.appId,
+      baseAppVersionId: (
+        currentAppInstallation.appVersion as { current: string }
+      ).current,
+    });
+  assertStatus(updateCandidates, 200);
+  return updateCandidates.data;
+}
+
+export async function getLatestAvailableTargetAppVersionForAppVersionUpgradeCandidates(
+  apiClient: MittwaldAPIV2Client,
+  appId: string,
+  baseAppVersionId: string,
+): Promise<AppVersion | undefined> {
+  const versions = await apiClient.app.listUpdateCandidatesForAppversion({
+    appId,
+    baseAppVersionId,
+  });
+  assertStatus(versions, 200);
+  if (versions.data.length === 0) {
+    return undefined;
+  }
+  let latestVersion = "0.0.0";
+  for (const version of versions.data) {
+    if (gt(version.internalVersion, latestVersion)) {
+      latestVersion = version.internalVersion;
+    }
+  }
+  return versions.data.find(
+    (item: AppVersion) => item.internalVersion === latestVersion,
+  );
```

Comment:
Can be simplified; just store the entire version object as `latestVersion` and save yourself the lookup at the end:

```suggestion
  let latestVersion: AppVersion|undefined;
  for (const version of versions.data) {
    if (gt(version.internalVersion, latestVersion?.internalVersion ?? "0.0.0")) {
      latestVersion = version;
    }
  }
  return latestVersion;
```